modules/storages/app/services/storages/one_drive_managed_folder_permissions_service.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  class OneDriveManagedFolderPermissionsService < BaseService
+    using Peripherals::ServiceResultRefinements
+
+    OP_PERMISSIONS = %i[read_files write_files create_files delete_files share_files].freeze
+
+    def self.i18n_key = "one_drive_sync_service"
+
+    class << self
+      def call(storage:, project_storages_scope: nil)
+        new(storage:, project_storages_scope:).call
+      end
+    end
+
+    def initialize(storage:, project_storages_scope: nil)
+      super()
+      @storage = storage
+      @project_storages = (project_storages_scope || @storage.project_storages).active.automatic
+    end
+
+    def call
+      with_tagged_logger([self.class.name, "storage-#{@storage.id}"]) do
+        apply_permission_to_folders
+
+        @result
+      end
+    end
+
+    private
+
+    # rubocop:disable Metrics/AbcSize
+    def apply_permission_to_folders
+      info "Setting permissions to project folders"
+      @project_storages.includes(:project)
+                                   .where.not(project_folder_id: nil)
```

Comment:
Here we miss the automatic and enabled checks.

Reply:
We don't, because I sneaked this into the initializer:

```ruby
@project_storages = (project_storages_scope || @storage.project_storages).active.automatic
```

Note how the parenthesis do the work for us here. I mostly switched to this pattern after realizing that some of the services use the scope more than once. Though I think my original implementation still applied this base-scope lazily.